apps/aehttp/test/aehttp_ws_test_utils.erl

```diff
@@ -0,0 +1,169 @@
+-module(aehttp_ws_test_utils).
+-behaviour(websocket_client).
+
+%% API
+-export([start_link/2,
+         send/4,
+         register_test_for_event/3,
+         unregister_test_for_event/3,
+         wait_for_event/2,
+         wait_for_event/3,
+         stop/1]).
+
+%% behaviour exports
+-export([
+         init/1,
+         onconnect/2,
+         ondisconnect/2,
+         websocket_handle/3,
+         websocket_info/3,
+         websocket_terminate/3
+        ]).
+
+-define(DEFAULT_EVENT_TIMEOUT, 5000).
+-define(DEFAULT_SUB_TIMEOUT, 1000).
+
+start_link(Host, Port) ->
+    WsAddress = "ws://" ++ Host ++ ":" ++ integer_to_list(Port) ++ "/websocket",
+    ct:log("connecting to ~p", [WsAddress]),
+    {ok, Pid} = websocket_client:start_link(WsAddress, ?MODULE, self()),
+    case wait_for_event(websocket, connected, ?DEFAULT_SUB_TIMEOUT) of
+        ok ->
+            {ok, Pid};
+        timeout ->
+            {error, timeout}
```

Comment:
When this happen, does WS client slow to connect (potentially for server that ran out of acceptors) stay lingering expecting to connect? Hence eventually consuming connection? Hence eventually potentially putting unexpected message in mailbox of test process?